compiler/next/include/chpl/uast/TypeConstructor.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_TYPE_CONSTRUCTOR_H
+#define CHPL_UAST_TYPE_CONSTRUCTOR_H
+
+#include "chpl/uast/Call.h"
+#include "chpl/queries/Location.h"
+
+namespace chpl {
+namespace uast {
+
+
+//
+// TODO: Better name for me besides TypeConstructor?
+//
```

Comment:
Hmm. The way we represent this in the old compiler is a call, with the called expression being at type. That doesn't seem like too much AST overloading to me because we already can have methods on types and functions returning types.

Reply:
So do we want to represent this as a straight up `FnCall`? I am hesitant to do that because it implies syntax that isn't there (?).

Reply:
Consider this program:

``` chapel
type t1 = atomic int;
type t2 = atomic(int);

writeln(t1:string);
writeln(t2:string);
```

IMO `atomic int` is a convenient way of writing `atomic(int)`. We could represent it as a `FnCall` (with a flag to distinguish it syntactically, if we care to) and I'd prefer that approach. Specifically for something like `atomic int` I'd expect we'd produce an `FnCall(calledExpr=Identifier("atomic), argument=Identifier("int"))` e.g.

For FnCalls generally, we don't know if the called thing is a function name or a type name, so we can't strict separation of the node types for the two.

My main concern about adding a uAST node for this case is that it doesn't cover all type construction calls or even a majority of them. It only covers those that are specifically called out in the parser. (While we could help that a bit with a different name than `TypeConstructor`, I think I'd still rather have them end up as `FnCall`).